src/Illuminate/Routing/Router.php

```diff
@@ -715,7 +715,9 @@ public function gatherRouteMiddleware(Route $route)
                 return true;
             }
 
-            $reflection = new ReflectionClass($name);
+            [$className] = explode(':', $name, 2);
```

Comment:
or: `Str::before()`

Reply:
woah, good idea :open_mouth:, let me try

Reply:
done,  thank you @mstaack